src/renderers/native/ReactNativeFiberInspector.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactNativeFiberInspector
+ * @flow
+ */
+'use strict';
+
+const ReactNativeComponentTree = require('ReactNativeComponentTree');
+const ReactFiberTreeReflection = require('ReactFiberTreeReflection');
+const getComponentName = require('getComponentName');
+
+import type {Fiber} from 'ReactFiber';
+
+if (__DEV__) {
+  var traverseOwnerTreeUp = function(hierarchy, instance: any) {
+    if (instance) {
+      hierarchy.unshift(instance);
+      traverseOwnerTreeUp(hierarchy, instance._debugOwner);
+    }
+  };
+
+  var getOwnerHierarchy = function(instance: any) {
+    var hierarchy = [];
+    traverseOwnerTreeUp(hierarchy, instance);
+    return hierarchy;
+  };
+
+  var lastNotNativeInstance = function(hierarchy) {
+    for (let i = hierarchy.length - 1; i > 1; i--) {
+      const instance = hierarchy[i];
+      if (!instance.viewConfig) {
+        return instance;
+      }
+    }
+    return hierarchy[0];
+  };
+
+  var getHostNode = function(fiber: Fiber, findNodeHandle) {
+    let hostNode;
+    // look for children first for the hostNode
+    // as composite fibers do not have a hostNode
+    while (fiber) {
+      hostNode = findNodeHandle(fiber.stateNode);
```

Comment:
We shouldn't even expose stateNode of HostComponent. That data structure will need to change and likely turn into something native. In fact, it may just be a number.